app/controllers/application_controller.rb

```diff
@@ -534,11 +534,16 @@ def fetch_user_from_params(opts = nil, eager_load = [])
     opts ||= {}
     user = if params[:username]
       username_lower = params[:username].downcase.chomp('.json')
```

Comment:
Not related to your change, but is the `chomp('.json')` really needed here? Looks weird.

Also, it probably should use `User.normalize_username(params[:username])` to prevent problems with Unicode.

Reply:
I'll defer to another PR for this :)